validator-flow.md

```diff
@@ -36,15 +38,54 @@ Attesting:
+    - Pre-Gloas forks: Max wait `SECONDS_PER_SLOT / 3` seconds into the assigned slot
+    - Post-Gloas forks: Max wait [ATTESTATION_DUE_BPS_GLOAS](https://github.com/ethereum/consensus-specs/blob/00d531949b1f30516979b60ddd2a411e7f388299/specs/gloas/validator.md#time-parameters) seconds into the assigned slot
 3. [Fetch AttestationData](#/ValidatorRequiredApi/produceAttestationData)
 4. [Submit Attestation](#/ValidatorRequiredApi/submitPoolAttestations) (AttestationData + aggregation bits)
     - Aggregation bits are `Bitlist` with length of committee (received in AttesterDuty)
     with bit on position `validator_committee_index` (see AttesterDuty) set to true
 5. If aggregator:
-    - Wait for `SECONDS_PER_SLOT * 2 / 3` seconds into the assigned slot
+    - Pre-Gloas forks: Wait for `SECONDS_PER_SLOT * 2 / 3` seconds into the assigned slot
+    - Post-Gloas forks: Wait for [AGGREGATE_DUE_BPS_GLOAS](https://github.com/ethereum/consensus-specs/blob/00d531949b1f30516979b60ddd2a411e7f388299/specs/gloas/validator.md#time-parameters) seconds into the assigned slot
     - [Fetch aggregated Attestation](#/ValidatorRequiredApi/getAggregatedAttestation) from Beacon Node you've subscribed to your subnet
     - [Publish SignedAggregateAndProofs](#/ValidatorRequiredApi/publishAggregateAndProofs)
 
 Monitor chain block reorganization events (TBD) as they could change attesters and aggregators.
 If reorg is detected, ask for new attester duties and proceed from 1..
+
+### PTC Attesting
+
+On start of every epoch beginning with the Gloas fork, validator should [fetch PTC duties](#/Validator/getPtcDuties) for epoch + 1.
+Result are array of objects with validator index and assigned slot for payload timeliness committee participation.
+
+PTC Attesting:
+1. Wait for new BeaconBlock for the assigned slot
+    - Max wait: `SECONDS_PER_SLOT / 4` seconds into the assigned slot
+2. [Fetch PayloadAttestationData](#/ValidatorRequiredApi/producePayloadAttestationData) for the assigned slot
+3. Sign PayloadAttestationData to create PayloadAttestationMessage
+4. [Submit PayloadAttestationMessage](#/ValidatorRequiredApi/submitPayloadAttestationMessage)
+    - Must be submitted by [PAYLOAD_ATTESTATION_DUE_BPS](https://github.com/ethereum/consensus-specs/blob/00d531949b1f30516979b60ddd2a411e7f388299/specs/gloas/validator.md#time-parameters) of slot duration
+    - Attestation indicates whether execution payload envelope has been seen for the block and if blobs were received
+
+Monitor chain block reorganization events (TBD) as they could change PTC assignments.
+If reorg is detected, ask for new PTC duties and proceed from 1..
+
+### Builder (Optional)
+
+Post-Gloas fork, validators may optionally act as builders to submit execution payload bids for block inclusion.
+This requires registering with builder-specific withdrawal credentials (`BUILDER_WITHDRAWAL_PREFIX`).
+
+Building:
+1. [Fetch ExecutionPayloadBid](#/Validator/getExecutionPayloadBid) from beacon node
```

Comment:
when would we actually ask the beacon node for a bid? currently the bidding for slot _N_ already starts in slot _N-1_